packages/@webex/plugin-cc/src/services/core/aqm-reqs.ts

```diff
@@ -0,0 +1,314 @@
+import {Signal} from './Signal'; // TODO: remove and use Event Emitters after Adhwaith PR is merged
+import {Msg} from './GlobalTypes';
+import * as Err from './Err';
+import {HTTP_METHODS, WebexRequestPayload} from '../../types';
+import HttpRequest from './HttpRequest';
+import LoggerProxy from '../../logger-proxy';
+import {
+  CbRes,
+  Conf,
+  ConfEmpty,
+  EvtConf,
+  EvtRes,
+  Pending,
+  Req,
+  Res,
+  ResEmpty,
+  Timeout,
+} from './types';
+
+export const TIMEOUT_REQ = 20000;
+const TIMEOUT_EVT = TIMEOUT_REQ;
+
+export class AqmReqs {
+  private pendingRequests: Record<string, Pending> = {};
+  private pendingEvents: Record<string, Pending> = {};
+  // pendingNotifCancelrequest is used for handling request cancel events based on another action. When a promise can be resolved using multiple events.
+  // example consult and cancelCtq(end in new API)  when we cancel consult to ques we need to resolve both consul and cancel ctq promises
+  // #CX-14258 : Consult to DN failing if Consult to Queue is cancelled in the first try #2344
+  private pendingNotifCancelrequest: Record<string, Pending> = {};
+  private httpRequest: HttpRequest;
+  constructor() {
+    this.httpRequest = HttpRequest.getInstance();
+    this.httpRequest.getWebSocket().on('event', (eventData) => {
+      LoggerProxy.logger.log(`Received event: ${eventData.type}`);
+      this.onMessage(eventData);
+    });
+  }
+
+  req<TRes, TErr, TReq>(c: Conf<TRes, TErr, TReq>): Res<TRes, TReq> {
+    return (p: TReq, cbRes?: CbRes<TRes>) => this.makeAPIRequest(c(p), cbRes);
+  }
+
+  reqEmpty<TRes, TErr>(c: ConfEmpty<TRes, TErr>): ResEmpty<TRes> {
+    return (cbRes?: CbRes<TRes>) => this.makeAPIRequest(c(), cbRes);
+  }
+
+  evt<T>(p: EvtConf<T>): EvtRes<T> {
```

Comment:
Cleanup event-specific code. It is not needed as we will have different event emission implemented

Reply:
addressed